rollup.config.js

```diff
@@ -0,0 +1,63 @@
+import typescript from '@rollup/plugin-typescript';
+import json from '@rollup/plugin-json';
+import shebang from '@walrus/rollup-plugin-shebang';
+import multiInput from 'rollup-plugin-multi-input';
+import fs from "fs";
+import commonjs from '@rollup/plugin-commonjs';
+import resolve from '@rollup/plugin-node-resolve';
+import pkge from './package.json'
+
+let filepaths = [];
+
+const walk = (root) => {
+  let currentPaths = fs.readdirSync(root);
+  for (let path of currentPaths) {
+    const abspath = `${root}/${path}`;
+    const isFile = /.*\.(ts|js|json)$/.test(path);
+
+    if (isFile) {
+      filepaths.push(abspath);
+    } else {
+      // isDirectory
+      walk(abspath);
+    }
+  }
+
+}
+
+walk('./src/commands');
+filepaths.push("./src/index.ts");
+
+// exclude type and test files
+filepaths = filepaths
+  .filter(path => !path.includes("test"))
+  .filter(path => !path.includes("type"));
+
+// console.log(filepaths);
+
+// exclude all dependencies from being bundled together, except for the problematic jsonLd, used by "open-attestation", "oa-verify" dependencies
+let deps = [...Object.keys(pkge.dependencies), ...Object.keys(pkge.devDependencies)]
+deps = deps.filter( (dep) => !(/oa\-|open-attestation/.test(dep)) )
+
+// console.log(deps);
+
+export default {
+  input: filepaths,
+  output: {
+    dir: 'rollup-build/cjs',
+    format: 'cjs',
+    exports: 'named'
+
+  },
+  external: [...deps],
```

Comment:
why need external? shouldn't it be handled by resolve plugin ? Unless it's specified in resolve plugin doc to use external?

Reply:
3rd-party dependencies, e.g. node modules, shouldn't be bundled together

https://www.mixmax.com/engineering/rollup-externals:
"if you open bundle.js [without specifying react as an external], you'll see something shocking: it contains the entirety of React and React DOM. That's 7000 LoC!

This occurs because, with the configuration above, React is no longer an external dependency: you've directed Rollup to bundle it alongside your application's local JavaScript. But there are critical differences between your application's JS and React's.
Why bundling 3rd-party dependencies can be a very bad idea..."

Reply:
The resolve plugin merely helps react locate the path of the module. By default, rollup does not know how to find node modules. For example, for the line, `import axios from "axios" `, rollup thinks that there is a
relative file called axios.js

Reply:
The question is about whether we should use external or rely on resolve option

Also react is completely irrelevant here as the comment is merely for library authors that shouldn't bundle dependencies, not our case here

Reply:
I tried with [resolveOnly option](https://github.com/rollup/plugins/tree/master/packages/node-resolve/#resolveonly)
```
resolve({  resolveOnly : [/^@govtechsg\/open-.*$/, "@govtechsg/open-attestation"]} ),
```
doesn't work. The esm bug still remains

Reply:
It's probably because it's not an issue with OA lib but with transitive dependencies. Try with the library causing issues instead (it's not json ld)